src/patch.ml

```diff
@@ -81,14 +81,16 @@ let rec apply_hunk ~cleanly ~fuzz (last_matched_line, offset, lines) ({mine_star
                 hunk
             in
             let hunk =
-              if Lib.List.last hunk.mine = (Lib.List.last hunk.their : string) then
+              let rev_mine = List.rev hunk.mine in
+              let rev_their = List.rev hunk.their in
+              if List.hd rev_mine = (List.hd rev_their : string) then
```

Comment:
Can we use String.equal explicitly here to avoid the potential polymorphic compare? I understand that the ` : string` should enable the OCaml compiler to use the right function, but I prefer to be explicit.